### PR TITLE
[Elastic Agent] Retry deamon reload during enroll

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

Adds retry for daemon reload 

## Why is it important?

https://github.com/elastic/fleet-server/issues/297

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Comment by ph** [comment]
@blakerouse Can you review this, @michalpristas I've moved to the right board :)

File: x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go [hunk 1]
```diff
@@ -323,6 +323,33 @@ func (c *enrollCmd) prepareFleetTLS() error {
...
 }
 
+func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
+	err := c.daemonReload(ctx)
+	signal := make(chan struct{})
+	backExp := backoff.NewExpBackoff(signal, 60*time.Second, 10*time.Minute)
+
+	for i := 5; i >= 0; i-- {
+		retry := false
+		if errors.Is(err, fleetapi.ErrTooManyRequests) {
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go:333]: These errors don't seem to apply to daemon reloading.

File: x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go [hunk 2]
```diff
@@ -323,6 +323,24 @@ func (c *enrollCmd) prepareFleetTLS() error {
 	return nil
 }
 
+func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
+	err := c.daemonReload(ctx)
+	signal := make(chan struct{})
+	backExp := backoff.NewExpBackoff(signal, 60*time.Second, 10*time.Minute)
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go:329]: Can we start with lower amounts? This seems rather large initial wait. Maybe lets do like 10 seconds to maximum of 1 minute?

This is a local daemon it should be quick to connect and communicate.

File: x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go [hunk 3]
```diff
@@ -323,6 +323,24 @@ func (c *enrollCmd) prepareFleetTLS() error {
 	return nil
 }
 
+func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
+	err := c.daemonReload(ctx)
```
Comment (blakerouse) [inline x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go:327]: If no error here then no reason to continue. Just return as it has been reloaded.

**Review by blakerouse — APPROVED** [review]
Looks good. Thanks for the fixes.

**Pull request merged**